### ISSUE TITLE
[metrics] Integrate metric calculation into Quaterion

### ISSUE THREAD

**Issue by generall** [qdrant/quaterion#61]
Allow user to specify "nearest-neighbor based accuracy metrics" in TrainableModel similar to loss functions.
 
Similar to https://kevinmusgrave.github.io/pytorch-metric-learning/testers/

Variant examples:
* For group samples - encode train dataset with current version of the model and evaluate if validation samples are properly classified by retrieval (rp@1)
* For pair sampler - encode one object of each pair and on the second run estimate precision@N

Perform metrics calculation during the eval epoch.
